art/defences/trainer/adversarial_trainer_trades_pytorch.py

```diff
@@ -240,7 +245,7 @@ def _batch_process(self, x_batch: np.ndarray, y_batch: np.ndarray) -> Tuple[floa
         )
 
         # Check label shape
-        if self._classifier._reduce_labels:  # pylint: disable=W0212
+        if self._classifier._reduce_labels and y_preprocessed.ndim > 1:  # pylint: disable=W0212
```

Comment:
This change is not needed because ART uses internally the one-hot-encoded format for classification labels.